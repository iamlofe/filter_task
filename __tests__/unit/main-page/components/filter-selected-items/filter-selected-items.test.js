import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';

import FilterSelectedItems from 'main-page/components/filter-selected-items/filter-selected-items';
import { getMockListItems } from '../../mock-store/mock-filter-reducer';

describe('account component', () => {
    const baseProps = {
        filteredList: getMockListItems(),
        selectedItems: new List([4, 3])
    };

    it('is shallow rendered without crashing', () => {
        const filterSelectedItems = <FilterSelectedItems {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        expect(filterSelectedItemsComponent).toMatchSnapshot();
    });

    it('should return truth length', () => {
        const filterSelectedItems = <FilterSelectedItems {...baseProps} />;

        const filterSelectedItemsComponent = shallow(filterSelectedItems);
        expect(filterSelectedItemsComponent.find('.filter-container__list-selected-items').children().length).toEqual(baseProps.selectedItems.count());
    });
});
