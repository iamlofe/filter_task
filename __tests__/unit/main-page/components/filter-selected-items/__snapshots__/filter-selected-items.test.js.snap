// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account component is shallow rendered without crashing 1`] = `
<div
  className="filter-container__list-selected-items"
>
  <p
    key="3"
  >
    results
  </p>
  <p
    key="4"
  >
    results2
  </p>
</div>
`;
